### ISSUE TITLE
Nodes Created When Querying Edges from Graph Editor

### ISSUE THREAD

**Issue by TropicalPenguin** [orientechnologies/orientdb#7491]
## Expected behavior  
SELECT should not mutate graph

## Actual behavior  
After creating two nodes (A, B), and two edges (A->B, B->B), and performing two SELECT commands inside the studio graph editor, on the second (and subsequent) SELECTs three new vertices are created, duplicating B.

## Steps to reproduce  

Create a new database, and set up the following schema:

```
CREATE CLASS foo EXTENDS V
CREATE CLASS bar EXTENDS V
CREATE CLASS barfoo EXTENDS E
CREATE CLASS barbar EXTENDS E
```

Run the following batch:

```
BEGIN
LET foo = CREATE VERTEX foo
LET bar = CREATE VERTEX bar
CREATE EDGE barfoo FROM $bar TO $foo
CREATE EDGE barbar FROM $bar TO $bar
LET result = SELECT $first AS tail,$last AS head FROM $foo LET first = in('barfoo')[0],last = $first.out('barbar')
COMMIT
RETURN $result
```

Then, in the graph editor (_this error only occurs from the graph editor_):

`SELECT FROM V`

The two vertices created in the batch appear.

`SELECT FROM E`

Three new vertices (of class 'bar') are added. From now on, this can be repeated, with three new vertices being added with each `SELECT FROM E` command.

Also, notably, no edges appear in the editor.

**Comment by luigidellaquila** [comment]
Hi @TropicalPenguin 

I'm working to a fix, I'll let you know soon

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @TropicalPenguin 

I just pushed a fix for this problem, you can find it in branch 2.2.x.
The fix will be released with 2.2.22

Please consider that edges created with the script you posted are broken, so you have to drop and re-create them to have a consistent db

Thanks

Luigi
